### PR TITLE
Gherkin Scritps for ODC-3924

### PR DESCRIPTION
**Pull request by gajanan-more**
Fixes: https://issues.redhat.com/browse/ODC-4043

Problem: Gherkin Scripts Design for Event Source sink to URI epic ODC-3924

Solution: These are the gherkin scripts designed covering all the scenarios related to Event Source sink to URI epic ODC-3924

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    And user will a node of URI
+    And user will see that event source is sinked with URI
+
+
+@regression
+Scenario: Context Menu for URI
+    Given user has sinked an event source to URI
+    And user is at Topology page
+    When user right-clicks on URI
+    Then user will see a a context menu opened for URI
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature:51]:
```suggestion
    Then user will see a context menu opened for URI 
```

File: frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    Then user will see a a context menu opened for URI 
+
+
+@regression
+Scenario: Edit URI option
+    Given user has sinked an event source to URI
+    And user is at Topology page
+    When user right-clicks on URI
+    And user clicks on Edit URI option
+    Then user will be redirected to edit the URI
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature:60]:
```suggestion
    Then user will be shown with modal to edit the URI
```

File: frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    User should be able to sink with resource as well as with URI
+
+
+Background: 
+    Given user has installed OpenShift Serverless Operator
+    And user is at Developer perspective
+
+
+@regression
+Scenario: Resource and URI radio button
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature:11]: Update the scenario in more descriptive way, something like Event Sources page details

File: frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+Scenario: Event Source sink to URI 
+    Given user is at +Add page
+    When user clicks on Event Sources card
+    And user selects an event source card
+    And user selects sink to URI option
+    And user enters the URI
+    And user fills other information
+    And user clicks on the Create button
+    Then user will be redirected to Topology page
+    And user will a node of URI
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature:42]: check the sentence once

File: frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    And user is at Topology page
+    When user right clicks on the event source
+    And user clicks on the Move Sink option
+    And user selects sink to Resource option
+    And user selects the same resource from Select Resource dropdown
+    Then user will see that save button is disabled
+
+
+@regression, @manual
+Scenario: Move sink from Resource to Resource
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature:150]: Duplicate scenario - same as above

Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature]: It's not duplicate. Updated the Scenario Description to understand it better

File: frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature (hunk 4 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature]: Updated the statement

File: frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature [hunk 6]
```diff
@@ -0,0 +1,169 @@
+Feature: Event Sources can able sink to URI as well as Resource
+    User should be able to sink with resource as well as with URI
```
Comment (sanketjecrc) [inline frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature:2]:
```suggestion
    User should be able to sink Event Sources with resource as well as with URI
```

Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/sink-to-URI-ODC-3924.feature:2]:
```suggestion
    User should be able to sink Event Sources with resource as well as with URI
```

**Pull request merged**